### PR TITLE
Rework of PR #10064 - Decoders and Rules update for Apache

### PR DESCRIPTION
**Pull request by 72nomada**
|Related issues|
|---|
| #10976 | 

## Description

This PR aims to resolve issues related to and detected under #10064 PR 

- Fixed and missing copyright headers.
- Missing software details.
- Fixed indentation issues.
- Fixed the rule descriptions.
- Added the rule ID range.
- Added rule ID 30113.
- [ ] Triple check - Modified current rule 30113 to 30319.
- Fixed group tag elements order.
- Fixed XML tag order.
- Removed sample logs from rule file.

### Syntax

- [x] 1.a Rule tags order must be compliant.
- [x] 1.b Decoder tags order must be compliant.
- [x] 1.c XML blocks must as compact as possible.
- [X] 1.d Only one empty line between rule/decoder and the next rule/decoder.
- [X] 1.e Decoder extracted fields names must use '_' whether a space is needed.
- [X] 1.f Decoder name must use '-' whether a space is needed.
- [X] 2.d XML and ini files must use correct indentation
 
### Grammar

- [X] 2.a Grammar quality.
- [X] 2.b Similar phrases must keep tenses and expressions.
- [X] 2.c Grammar basic rules like capitalization, punctuation marks, sentence construction, and others.

### Semantic

- [x] 3.a New decoder, rule, or test are written in the correct file and grouped correctly inside the file.
- [x] 3.b Group similar rules under same ID's group.
- [x] 3.c: 
     - [x] 3.c1 Find and reuse group name before creating a new one.
     - [x] 3.c2 Include a new group definition in a PR comment.
     - [x] 3.c3 Any group name must use '_' whether it does need a space char.
     - [x] 3.c4 The groups inside a tag must be sorted in alphabetical order.
- [x] 3.d Rule level should be compliant with the [documentation](https://documentation.wazuh.com/current/user-manual/ruleset/rules-classification.html).

### Unit testing

- [X] 4.a A metadata block at the beginning of the .ini file describing software, version, and logs source.
- [X] 4.b Each new rule must have at least one test entry in the correct .ini file.
- [x] 4.c Runtest.py must pass and must include the results in raw format here.

<details><summary>runtest.py results</summary>
<p>

```
python runtests.py
- [ File = ./tests/opensmtpd.ini ] ---------
.......

- [ File = ./tests/arbor.ini ] ---------
..

- [ File = ./tests/aws_s3_access.ini ] ---------
.......

- [ File = ./tests/panda_paps.ini ] ---------
........

- [ File = ./tests/fortigate.ini ] ---------
........................................

- [ File = ./tests/iptables.ini ] ---------
........

- [ File = ./tests/vuln_detector.ini ] ---------
..

- [ File = ./tests/syslog.ini ] ---------
......

- [ File = ./tests/fortiauth.ini ] ---------
....

- [ File = ./tests/postfix.ini ] ---------
..

- [ File = ./tests/pam.ini ] ---------
.....

- [ File = ./tests/vsftpd.ini ] ---------
....

- [ File = ./tests/audit_recon.ini ] ---------
..

- [ File = ./tests/kernel_usb.ini ] ---------
......

- [ File = ./tests/sudo.ini ] ---------
........

- [ File = ./tests/systemd.ini ] ---------
..

- [ File = ./tests/audit_lateral.ini ] ---------
......

- [ File = ./tests/fireeye.ini ] ---------
...

- [ File = ./tests/freepbx.ini ] ---------
......

- [ File = ./tests/f5_big_ip.ini ] ---------
...........................................

- [ File = ./tests/cylance.ini ] ---------
.......

- [ File = ./tests/netscreen.ini ] ---------
....

- [ File = ./tests/huawei_usg.ini ] ---------
...

- [ File = ./tests/junos.ini ] ---------
...

- [ File = ./tests/cpanel.ini ] ---------
.......

- [ File = ./tests/mcafee_epo.ini ] ---------
.

- [ File = ./tests/glpi.ini ] ---------
...

- [ File = ./tests/api.ini ] ---------
............................

- [ File = ./tests/rsh.ini ] ---------
..

- [ File = ./tests/nginx.ini ] ---------
............

- [ File = ./tests/audit_scp.ini ] ---------
.

- [ File = ./tests/cloudlfare-waf.ini ] ---------
.............

- [ File = ./tests/sophos-utm-firewall.ini ] ---------
......

- [ File = ./tests/icinga.ini ] ---------
....

- [ File = ./tests/checkpoint_smart1.ini ] ---------
..................

- [ File = ./tests/doas.ini ] ---------
....

- [ File = ./tests/sophos.ini ] ---------
........

- [ File = ./tests/web_rules.ini ] ---------
..........

- [ File = ./tests/ossec.ini ] ---------
.....

- [ File = ./tests/sophos_fw.ini ] ---------
..........

- [ File = ./tests/openldap.ini ] ---------
.........

- [ File = ./tests/nextcloud.ini ] ---------
.......

- [ File = ./tests/cimserver.ini ] ---------
..

- [ File = ./tests/paloalto.ini ] ---------
................

- [ File = ./tests/unbound.ini ] ---------


- [ File = ./tests/cisco_ios.ini ] ---------
.................

- [ File = ./tests/su.ini ] ---------
.....

- [ File = ./tests/oscap.ini ] ---------
................................

- [ File = ./tests/gitlab.ini ] ---------
.........................

- [ File = ./tests/gcp.ini ] ---------
...........

- [ File = ./tests/github.ini ] ---------
....................................................................................................................................................................................................................................................................................................................................

- [ File = ./tests/exim.ini ] ---------
.......

- [ File = ./tests/named.ini ] ---------
.....

- [ File = ./tests/samba.ini ] ---------
....

- [ File = ./tests/owlh.ini ] ---------
....

- [ File = ./tests/pix.ini ] ---------
......................

- [ File = ./tests/modsecurity.ini ] ---------
......

- [ File = ./tests/apparmor.ini ] ---------
.....

- [ File = ./tests/proftpd.ini ] ---------
.......

- [ File = ./tests/eset.ini ] ---------
........

- [ File = ./tests/firewalld.ini ] ---------
..

- [ File = ./tests/SonicWall.ini ] ---------
........

- [ File = ./tests/cisco_ftd.ini ] ---------
..........................................

- [ File = ./tests/trendmicro-cloud-one.ini ] ---------
.................

- [ File = ./tests/dovecot.ini ] ---------
...............

- [ File = ./tests/sysmon.ini ] ---------
...

- [ File = ./tests/auditd.ini ] ---------
...

- [ File = ./tests/mailscanner.ini ] ---------
.

- [ File = ./tests/openvpn_ldap.ini ] ---------
..

- [ File = ./tests/sshd.ini ] ---------
...........................................

- [ File = ./tests/pfsense.ini ] ---------
..

- [ File = ./tests/squid_rules.ini ] ---------
..

- [ File = ./tests/office365.ini ] ---------
................................................................................................................................

- [ File = ./tests/exchange.ini ] ---------
..

- [ File = ./tests/cisco_asa.ini ] ---------
........................................................................................

- [ File = ./tests/web_appsec.ini ] ---------
...............................

- [ File = ./tests/dropbear.ini ] ---------
...

- [ File = ./tests/apache.ini ] ---------
..................
```
</p>
</details>

- [X] 4.d New CDB lists must include the proper test entry in the correct .ini file.

### E2E testing

- ~~5.a Logs for new or modified decoder/rules sent to a manager running with this PR ruleset appear in Kibana.~~ 
- ~~5.b There are not affected or broken visualizations on Kibana.~~
- ~~5.c New or modified items can be seen correctly using APP ruleset navigation.~~

### Elasticsearch Template

- [X] 6.a Known fields with output format managed and usually used for searching are included in template array index.query.default_field. 
- [X] 6.b The new field with the correct date format is stored as a "date" type field in the template.
- [X] 6.c The new extracted IP fields are in the pipeline as "geo" field and "geo_point" type in the template. 
- [X] 6.d Known fields with output format managed and usually used for searching are included in the template.

### Stoppers

- [x] 7.a No previous rule ID changes without triple check.
- [x] 7.b No previous decoder name changes without triple check.

<details><summary>No rule is using old decoder name</summary>
<p>

```
# grep -inH 'apache24-errorlog-ip' /var/ossec/ruleset/rules/*.xml
# grep -inH 'apache24-errorlog-ip-port' /var/ossec/ruleset/rules/*.xml
# grep -inH 'apache24-errorlog-glpi' /var/ossec/ruleset/rules/*.xml
#
```
</p>
</details>

<details><summary>No decoder is using old decoder name</summary>
<p>

```
# grep -inH 'apache24-errorlog-ip' /var/ossec/ruleset/decoders/*.xml
# grep -inH 'apache24-errorlog-ip-port' /var/ossec/ruleset/decoders/*.xml
# grep -inH 'apache24-errorlog-glpi' /var/ossec/ruleset/decoders/*.xml
#
```
</p>
</details>

- [x] 7.c No previous file name changes without triple check.
- [x] 7.d No previous test changes its 'rule' field value without triple check.

### Others

- [x] 8.a Each file has the correct copyright block.
- [x] 8.b The copyright block doesn't have "Author" only "Created by Wazuh". To include an "Author" request triple check.
- [x] 8.c The copyright block doesn't use "-". 
- [x] 8.d The rule files don't have any sample log.
- [x] 8.e The decoder file has sample logs next to the decoder that matches that log.
- [x] 8.f The decoder and rule files have information about software, version, and any helpful information.
- [x] 8.g The PR has a single commit with CHANGELOG changes in the correct format.
- [x] 8.h The new rule ID is not in use.
- [x] 8.i The new rule ID must be in the defined IDs range. 
- [x] 8.j New rules ID range must be verified with a triple check and noted in the rules ID document.
- [x] 8.k The new extracted IP fields are in the pipeline as "geo" field and "geo_point" type in the template.

### CODE REVIEW

File: ruleset/testing/tests/apache.ini [hunk 1]
```diff
@@ -22,57 +30,57 @@ rule = 30106
 alert = 5
 decoder = apache-errorlog
 
-[Code Red attack]
+[Code Red attack.]
```
Comment (Spades0) [inline ruleset/testing/tests/apache.ini:33]: Code Red Attack, I think.

File: ruleset/testing/tests/apache.ini [hunk 2]
```diff
@@ -22,57 +30,57 @@ rule = 30106
...
 
-[Attempt to access an non-existent file]
+[Attempt to access an non-existent file.]
 log 1 pass = [error] [client 66.31.142.16] File does not exist: /var/www/html/default.ida
 rule = 30112
 alert = 0
 decoder = apache-errorlog
 
-[Apache notice messages grouped]
+[Apache notice messages grouped.]
```
Comment (Spades0) [inline ruleset/testing/tests/apache.ini:45]: Apache: Notice messages grouped.

File: ruleset/testing/tests/apache.ini [hunk 3]
```diff
@@ -22,57 +30,57 @@ rule = 30106
...
 log 1 pass = [notice] Apache configured
 rule = 30103
 alert = 0
 decoder = apache-errorlog
 
-[Apache 2.2 error messages grouped]
+[Apache 2.2 error messages grouped.]
```
Comment (Spades0) [inline ruleset/testing/tests/apache.ini:51]: Apache 2.2: Error messages grouped. and so on

File: ruleset/rules/0250-apache_rules.xml [hunk 4]
```diff
@@ -213,34 +220,34 @@
...
+    <group>access_denied,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_SA.11,nist_800_53_AU.14,nist_800_53_AC.7,pci_dss_6.5.8,pci_dss_10.2.4,tsc_CC6.6,tsc_CC7.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
   </rule>
 
   <rule id="30307" level="6">
     <if_sid>30301</if_sid>
     <id>AH00550</id>
-    <description>Apache: Client sent malformed Host header. Possible Code Red attack.</description>
     <info type="link">http://www.cert.org/advisories/CA-2001-19.html</info>
     <info type="text">CERT: Advisory CA-2001-19 "Code Red" Worm Exploiting Buffer Overflow In IIS Indexing Service DLL</info>
+    <description>Apache: Client sent malformed Host header. Possible Code Red attack.</description>
```
Comment (xrisbarney) [inline ruleset/rules/0250-apache_rules.xml:231]: ~~Apache: Client sent malformed Host header. Possible Code Red attack.~~
Apache: Client sent malformed host header. Possible code red attack

File: ruleset/rules/0250-apache_rules.xml [hunk 5]
```diff
@@ -213,34 +220,34 @@
...
-    <group>access_denied,pci_dss_6.5.8,pci_dss_10.2.4,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_SA.11,nist_800_53_AU.14,nist_800_53_AC.7,tsc_CC6.6,tsc_CC7.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
+    <group>access_denied,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_SA.11,nist_800_53_AU.14,nist_800_53_AC.7,pci_dss_6.5.8,pci_dss_10.2.4,tsc_CC6.6,tsc_CC7.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
   </rule>
 
   <rule id="30307" level="6">
     <if_sid>30301</if_sid>
     <id>AH00550</id>
-    <description>Apache: Client sent malformed Host header. Possible Code Red attack.</description>
     <info type="link">http://www.cert.org/advisories/CA-2001-19.html</info>
     <info type="text">CERT: Advisory CA-2001-19 "Code Red" Worm Exploiting Buffer Overflow In IIS Indexing Service DLL</info>
```
Comment (xrisbarney) [inline ruleset/rules/0250-apache_rules.xml:230]: Is this info the direct title from CERT? Do we need to put it in correct sentence case and add a period at the end?

**Review by xrisbarney — CHANGES_REQUESTED** [review]

**Pull request closed without merging**